share-coordinator/src/main/java/org/apache/kafka/coordinator/share/metrics/ShareCoordinatorMetrics.java

```diff
@@ -153,4 +158,31 @@ public void record(String sensorName) {
             globalSensors.get(sensorName).record();
         }
     }
+
+    public void recordPrune(double value, TopicPartition tp) {
+        pruneMetrics.computeIfAbsent(tp, k -> new ShareGroupPruneMetrics(tp))
+            .pruneSensor.record(value);
+    }
+
+    private class ShareGroupPruneMetrics {
+        private final Sensor pruneSensor;
+
+        ShareGroupPruneMetrics(TopicPartition tp) {
+            String sensorNameSuffix = tp.toString();
+            Map<String, String> tags = Map.of(
+                "topic", tp.topic(),
+                "partition", Integer.toString(tp.partition())
+            );
+
+            pruneSensor = metrics.sensor(SHARE_COORDINATOR_STATE_TOPIC_PRUNE_SENSOR_NAME + sensorNameSuffix);
+
+            pruneSensor.add(
+                metrics.metricName("last-pruned-offset",
+                    METRICS_GROUP,
+                    "The offset at which the share-group state topic was last pruned.",
+                    tags),
+                new Value()
+            );
+        }
+    }
```

Comment:
Do we really need this or can rely on `metrics.getSensor(name)` as internally a map already exists there?

Reply:
@apoorvmittal10 
We can use that but we would anyway need a set or map to store sensor names since we need to remove the sensors from metrics in the `close` method. This is necessary as we do not know beforehand how many tps are there.

Reply:
No we needn't remove metrics explicitly since `removeSensor` method in `Metrics.java` https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java#L442 manages that internally.